client/src/hooks/useSession.ts

```diff
@@ -59,11 +62,14 @@ export const useSession = () => {
     if (toLogout) {
       destroySession()
         .then(() => refetch())
-        .then(() => apollo.resetStore());
+        .then(() => apollo.resetStore())
+        .catch((e) => setError(e));
```

Comment:
The weird thing about `fetch` is that it doesn't throw errors if there are issues with the request. That means even if we get a status 500 from the server, we don't get to the catch.  You can easily test this by calling `next('message')` in the `/login` middleware.

To have something to catch, you can simply throw an error inside create/destroySession if `ok` is falsy. You _could_ handle it it with a conditional in the `then`, but catching seems fine here!

Reply:
This is interesting. I generally was putting a hammer on server, by deleting some line in `app.ts` so it would error after another compilation. For that case `fetch` is indeed throwing.

Reply:
You may be seeing something different, but for what it's worth I was confused by the network errors appearing in the logs. They're not JS errors, they just register that something went wrong during a network request. If they go away when you check "Hide network" in the browser's dev console, we're talking about the same thing.

To the best of my knowledge, you can't catch those errors. However, if you're able to create JS errors by messing with the server, that's really interesting. At least to me, because I'll have learned a thing!

Reply:
Try these steps to reproduce:
- Remove `catch` chained to `destroySession()` and `createSession()`.
- While having running server, change something in `/server/src/app.ts` to make it not compile anymore. For example remove line with `const PORT = process.env.PORT || 5000;`
- Now when pressing _Login_ or _Logout_ there will be unhandled error, from the `fetch`.

Reply:
Thanks for the quick steps to reproduce, it does indeed error in that case! I did a little digging and it turns out that's just how it works: https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API#differences_from_jquery

Bit strange, but your code handles both cases, so I think we're good.